libcperciva/datastruct/seqptrmap.c

```diff
@@ -79,12 +79,19 @@ seqptrmap_add(struct seqptrmap * M, void * ptr)
 void *
 seqptrmap_get(struct seqptrmap * M, int64_t i)
 {
-	size_t pos = i - M->offset;
+	size_t pos;
+	int64_t pos_i;
 
 	/* Sanity check, but we cannot fail, so don't use an assert. */
 	if (i < 0)
 		return (NULL);
 
+	/* Check that the position will fit into size_t. */
+	pos_i = i - M->offset;
+	if ((pos_i < 0) || ((uint64_t)pos_i >= (uint64_t)SIZE_MAX))
```

Comment:
Casting down to an unsigned integer type is not undefined behaviour.  You only get UD if there's an overflow when casting to a *signed* integer type.